### ISSUE TITLE
Problems with character encoding 

### ISSUE THREAD

**Issue by jeanmile** [outr/mailgun4s#2]
Hi,
I have some troubles for sending email with accents characters like "éàè".
Have you any ideas how to solve this ?
Thanks

**Comment by darkfrog26** [comment]
Can you clarify what troubles you're having?  Is Mailgun rejecting?  Is the message never arriving?  Is the data getting corrupted?  Is an exception being thrown?

**Comment by jeanmile** [comment]
The message output is malformed.

Code used
```scala
def send() = {
    val mailgun: Mailgun = new Mailgun(s"my_domain", API_KEY)
    mailgun.send(Message.simple(
      from = EmailAddress("nobody@example.com", "Test App"),
      to = EmailAddress("jeanmile.le@gmail.com", "Joe User"),
      "Mailgun4s Rules!",
      html = "ééèè"
    ))
  }
```

Log 
```
[debug] o.a.http.wire - http-outgoing-0 >> "Test App <nobody@example.com>[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "--vGVQmbvypwrTAi3iLBgbCkb3UXkViRQ[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Disposition: form-data; name="subject"[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Type: text/plain[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Transfer-Encoding: 8bit[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Mailgun4s Rules![\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "--vGVQmbvypwrTAi3iLBgbCkb3UXkViRQ[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Disposition: form-data; name="to"[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Type: text/plain[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Transfer-Encoding: 8bit[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Joe User <jeanmile.le@gmail.com>[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "--vGVQmbvypwrTAi3iLBgbCkb3UXkViRQ[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Disposition: form-data; name="html"[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Type: text/plain[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "Content-Transfer-Encoding: 8bit[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "[\r][\n]"
HERE --> [debug] o.a.http.wire - http-outgoing-0 >> "????[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 >> "--vGVQmbvypwrTAi3iLBgbCkb3UXkViRQ--[\r][\n]"
[debug] o.a.h.i.n.c.InternalIODispatch - http-outgoing-0 [ACTIVE] Request ready
[debug] o.a.h.i.n.c.ManagedNHttpClientConnectionImpl - http-outgoing-0 192.168.1.80:49859<->34.208.124.23:443[ACTIVE][r:w][ACTIVE][r][NOT_HANDSHAKING][0][0][0][0]: Event cleared [w]
[debug] o.a.h.i.n.c.ManagedNHttpClientConnectionImpl - http-outgoing-0 192.168.1.80:49859<->34.208.124.23:443[ACTIVE][r:r][ACTIVE][r][NOT_HANDSHAKING][0][0][0][0]: 477 bytes read
[debug] o.a.http.wire - http-outgoing-0 << "HTTP/1.1 200 OK[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Access-Control-Allow-Headers: Content-Type, x-requested-with[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Access-Control-Allow-Origin: *[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Access-Control-Max-Age: 600[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Content-Disposition: inline[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Content-Type: application/json[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Date: Mon, 09 Oct 2017 18:03:50 GMT[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Server: nginx[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Content-Length: 114[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "Connection: keep-alive[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "[\r][\n]"
[debug] o.a.http.wire - http-outgoing-0 << "{[\n]"
[debug] o.a.http.wire - http-outgoing-0 << "  "id": "<20171009180350.126399.D9636C3F8FF1BB28@my_domain>",[\n]"
[debug] o.a.http.wire - http-outgoing-0 << "  "message": "Queued. Thank you."[\n]"
[debug] o.a.http.wire - http-outgoing-0 << "}"
[debug] o.a.h.headers - http-outgoing-0 << HTTP/1.1 200 OK
```

Mailgun received
```
Date: Mon, 09 Oct 2017 18:09:17 +0000
Content-Transfer-Encoding: 7bit
Received: by luna.mailgun.net with HTTP; Mon, 09 Oct 2017 18:03:50 +0000
Message-Id: <20171009180350.126399.D9636C3F8FF1BB28@my_domain>
To: Joe User <jeanmile.le@gmail.com>
From: Test App <nobody@example.com>
Subject: Mailgun4s Rules!
Content-Type: text/html; charset="ascii"
Mime-Version: 1.0

????
```

**Comment by darkfrog26** [comment]
Hmmm, it looks like it's setting the `charset` to `ascii`.  I'll have to see if I can change that to utf-8.  I'll get back to you.

**Comment by darkfrog26** [comment]
I've followed every reference to forcing charset encoding I can find and I'm still seeing the same problem you are.  I've submitted a ticket to Mailgun asking for advice on how to properly handle this and I'll let you know once they've responded.

**Comment by jeanmile** [comment]
I tried the original lib https://github.com/Nycto/mailgun-scala and it's working.

MailGun : 
```
Date: Mon, 09 Oct 2017 20:33:19 +0000
Content-Transfer-Encoding: quoted-printable
Received: by luna.mailgun.net with HTTP; Mon, 09 Oct 2017 16:26:21 +0000
Message-Id: <20171009162621.107172.2452AB63C332F7AB@ my_domain>
To: jeanmile.le@gmail.com
From: Test App <nobody@example.com>
Subject: Subject
Content-Type: text/html; charset="utf-8"
Mime-Version: 1.0
```

**Comment by darkfrog26** [comment]
@jeanmile that's useful...I'll take a look and see what they're doing differently.

**Comment by darkfrog26** [comment]
@jeanmile okay, I found the distinction: https://github.com/Nycto/mailgun-scala/blob/master/src/main/scala/Requestor.scala#L90

I didn't realize that I needed to set the Charset per part in the multipart.  Just tested and this fixes the problem.  I'll close this ticket once the code has been committed a new release made.  Sorry for the inconvenience.

**Comment by jeanmile** [comment]
I also found your fix https://github.com/outr/youi/commit/a16a68ec931a4d66720b35448fdd7e9c7ab88456.
It's work ! :thumbsup:

**Comment by darkfrog26** [comment]
Okay, I released 1.0.7 that fixes this.  Let me know if you have any further trouble.
